File: ClasseHeroi/Heroi.py
class Heroi:
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        if self.tipo == "mago":
            ataque = "magia"
        elif self.tipo == "guerreiro":
            ataque = "espada"
        elif self.tipo == "monge":
            ataque = "artes marciais"
        elif self.tipo == "ninja":
            ataque = "shuriken"
        else:
            ataque = "um ataque desconhecido"

        print(f"o {self.tipo} atacou usando {ataque}")


# Exemplo de uso
heroi1 = Heroi("Merlin", 100, "mago")
heroi1.atacar()

File: JavaJogoDoHeroi/heroi/src/Heroi.java
public class Heroi {
    private String nome;
    private int idade;
    private String tipo;

    public Heroi(String nome, int idade, String tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }

    public void atacar() {
        String ataque;
        switch (tipo.toLowerCase()) {
            case "mago":
                ataque = "magia";

                break;
            case "guerreiro":
                ataque = "espada";
                break;
            case "monge":
                ataque = "artes marciais";
                break;
            case "ninja":
                ataque = "shuriken";
                break;

            default:
                ataque = "um ataque desconhecido";
        }
        System.out.println("o " + tipo + " atacou usando " + ataque);
    }
    public static void main(String[] args) {
        Heroi heroi1 = new Heroi("Merlin", 100,"mago");
        heroi1.atacar();
    }

}
